common/include/pcl/pcl_macros.h

```diff
@@ -363,3 +363,9 @@ aligned_free (void* ptr)
 #else
   #define PCL_FALLTHROUGH ;
 #endif
+
+#if __cplusplus >= 201703L
```

Comment:
Please use the [feature test macro](https://en.cppreference.com/w/cpp/feature_test) as that is more descriptive.